File: Java/prog_lab_7/src/prog_lab_7/point.java
package prog_lab_7;
import java.util.Scanner;

public class point implements inputoutput , Cloneable{
	protected double x, y;//модификатор protected 
	protected String description;//модификатор protected
	
	public point() {
		description = new String();
		x = y = 0;
	}
	
	public point(double x) {
		description = new String();
		this.x = this.y = x;
	}
	
	public point(double x, double y) {
		description = new String();
		this.x = x;
		this.y = y;
	}
	
	public void vvod() {
		Scanner in = new Scanner(System.in);
		this.x = in.nextDouble();
		this.y = in.nextDouble();
	}
	
	public void vivod()	{
		System.out.printf("x=%.3f; y=%.3f", this.x, this.y);
	}

	public double distance(point sdpoint) {
		return Math.sqrt(Math.pow(this.x - sdpoint.x, 2) + Math.pow(this.y - sdpoint.y, 2));
	}
	
	public double getx() {
		return this.x;
	}
	
	public double gety() {
		return this.y;
	}
	
	public void init(double x, double y) {
		this.x = x;
		this.y = y;
		return;
	}
	
	public void descrin() {
		Scanner in = new Scanner(System.in);
		description = in.nextLine();
	}
	
	public void descrout() {
		System.out.println(description);
	}
	
	public int sum() {
		return (int)(x+y);
	}

	@Override
	public String toString() {//метод toString 
		return "point [x=" + x + ", y=" + y + ", description=" + description + "]";
	}	
	
	@Override
	protected point clone() throws CloneNotSupportedException{
		return (point) super.clone();
	}
}

File: C++/prog_lab_4/point.cpp
#define _CRT_SECURE_NO_WARNINGS
#include "point.h"
#include <iostream>
#include <math.h>

point::point()
{
	x = y = 0;
}

point::point(float x)
{
	this->x = this->y = x;
}

point::point(float x, float y)
{
	this->x = x;
	this->y = y;
}

void point::vvod()
{
	while (scanf("%f%f", &x, &y) < 2) {
		while (getchar() != '\n');
		printf("Координаты точки введены неверно.\n");
	}
	return;
}

void point::vivod()
{
	printf("x=%.3f, y=%.3f", x, y);
}

float point::distance(point* sdpoint)
{
	return sqrt(pow(sdpoint->x - x, 2) + pow(sdpoint->y - y, 2));
}

float point::getx()
{
	return x;
}

float point::gety()
{
	return y;
}

void point::descrin()
{
	std::cin >> description;
}

void point::descrout()
{
	std::cout << description;
}

void point::init(float x, float y)
{
	this->x = x;
	this->y = y;
}

void point::totalvvod()
{
	this->vvod();//вызов виртуальной функции не виртуальной функцией базового класса
	this->descrin();
}

point point::operator+(point b)
{
	point c;
	c.x = x + b.x;
	c.y = y + b.y;
	return c;
}

point point::operator++(int)
{
	x++;
	return point();
}

point point::operator++()
{
	y++;
	return point();
}

ostream& operator<<(ostream& s, const point& pt)
{
	s << "x = " << pt.x << ", y = " << pt.y;
	return s;
}

void pround(point& point)
{
	point.x = round(point.x);
	point.y = round(point.y);
}
